packages/p2p/src/components/my-profile/my-profile.jsx

```diff
@@ -1,70 +1,308 @@
+    }, []);
 
-Cell.propTypes = {
-    dimension: PropTypes.string,
-    lower_sub_value: PropTypes.string,
-    title: PropTypes.string,
-    upper_sub_value: PropTypes.string,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-};
+    React.useEffect(() => {
+        if (is_mounted) {
+            advertiserAccountStatus();
+            advertiserStats();
+            advertiserInfo();
+        }
+    }, [is_mounted]);
+
+    const advertiserAccountStatus = () => {
+        return new Promise(resolve => {
+            requestWS({
+                get_account_status: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { get_account_status } = response;
+                    const { authentication } = get_account_status;
+                    setHasPoa(!(authentication.document && authentication.document.status === 'none'));
+                    setHasPoi(!(authentication.identity && authentication.identity.status === 'none'));
+                }
+                resolve();
+            });
+        });
+    };
+
+    const advertiserInfo = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_info: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { p2p_advertiser_info } = response;
+                    setAdvertiserInfo(p2p_advertiser_info);
+                    setNickname(p2p_advertiser_info.name);
+                    setIsLoading(false);
+                }
+                resolve();
+            });
+        });
+    };
+
+    const advertiserStats = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_stats: 1,
+                id,
+            }).then(response => {
+                const { p2p_advertiser_stats } = response;
+                setStats(p2p_advertiser_stats);
+                resolve();
+            });
+        });
+    };
+
+    const handleSubmit = values => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_update: 1,
+                contact_info: values.contact_info,
+                payment_info: values.payment_info,
+                default_advert_description: values.default_advert_description,
+            });
+            resolve();
+        });
+    };
+
+    const validateForm = values => {
+        const validations = {
+            contact_info: [v => textValidator(v)],
+            default_advert_description: [v => textValidator(v)],
+            payment_info: [v => textValidator(v)],
+        };
+
+        const mapped_key = {
+            contact_info: localize('Contact details'),
+            default_advert_description: localize('Instructions'),
+            payment_info: localize('Payment details'),
+        };
+
+        const errors = {};
+
+        const error_messages = field_name => [
```

Comment:
Since it's a function, please use camelCase.

```suggestion
        const getErrorMessages = field_name => [
```